### ISSUE TITLE
Mask mode missing from Light2D

### ISSUE THREAD

**Issue by aledraken** [godotengine/godot#98790]
### Tested versions

Reproducible in any 4.0 version

### System information

Windows 10 - Godot 4.3

### Issue description

Light2D nodes no longer have "mask" as one of their blend modes like they did in previous versions, making creating Line Of Sights/Field Of Views in 2D impossible. Some say to use CanvasItems/Groups but there is 0 documentation/tutorials about this subject so I'm not even sure if it's actually possible to get the same effect.

### Steps to reproduce

Create a PointLight2D node
Try to change blend mode to mask
Notice that only add, substract and mix modes are available
Explode

### Minimal reproduction project (MRP)

N/A

**Comment by AThousandShips** [comment]
What functionality are you missing? In what way is this not possible in 4.x? Does this not work in this [example](https://github.com/godotengine/godot-demo-projects/tree/master/2d/light2d_as_mask)?

**Comment by aledraken** [comment]
> What functionality are you missing? In what way is this not possible in 4.x? Does this not work in this [example](https://github.com/godotengine/godot-demo-projects/tree/master/2d/light2d_as_mask)?

Hi, I'm trying to make a line of sight kinda like Darkwood/Zero sievert/Among Us, in the project you've sent they use a canvas material with light only, but that will show enemies and everything else also when they are not lit by the player, for example by any other light source.

Most of the workarounds I've found online and some I've created myself are too expensive and complicated for larger scenes and are generally too complex for an issue that didn't even exist in godot 3.

**Comment by AThousandShips** [comment]
Have you tested the new `Clip Children` mode? This is intended to be the replacement as far as I can find, see:
* https://github.com/godotengine/godot/issues/70358

For documenting the changes, please see:
* https://github.com/godotengine/godot-docs/issues/5121

If this is still not what you need, please open a [proposal](https://github.com/godotengine/godot-proposals) as this is for bugs, not missing features (even if they were removed)

Thank you for your report!

**Comment by aledraken** [comment]
> Have you tested the new `Clip Children` mode? This is intended to be the replacement as far as I can find, see:
> 
> * [blend mode mask is missing in Light2D node #70358](https://github.com/godotengine/godot/issues/70358)
> 
> For documenting the changes, please see:
> 
> * [Documentation pages to add or update for Godot 4.0 godot-docs#5121](https://github.com/godotengine/godot-docs/issues/5121)
> 
> If this is still not what you need, please open a [proposal](https://github.com/godotengine/godot-proposals) as this is for bugs, not missing features (even if they were removed)
> 
> Thank you for your report!

Yes, I've tried using clip children with subviewports but it's really heavy with larger scenes and kinda of a weird setup, I guess I'll open a proposal, thanks for your time.

**Comment by AThousandShips** [comment]
Oh my bad you didn't mention clip children in your report so didn't know you had tried that

**Comment by aledraken** [comment]
> Oh my bad you didn't mention clip children in your report so didn't know you had tried that

Np, btw if you'd like to take a look at the way I implemented it I've added my project in the proposal https://github.com/godotengine/godot-proposals/issues/11082
